ChatQnA/docker_compose/intel/hpu/gaudi/compose.yaml

```diff
@@ -76,9 +79,10 @@ services:
       HABANA_VISIBLE_DEVICES: all
       OMPI_MCA_btl_vader_single_copy_mechanism: none
       MAX_WARMUP_SEQUENCE_LENGTH: 512
-    command: --model-id ${RERANK_MODEL_ID} --auto-truncate
+    command: --model-id ${RERANK_MODEL_ID} --auto-truncate --otlp-endpoint $OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
   tgi-service:
     image: ghcr.io/huggingface/tgi-gaudi:2.0.6
+    privileged: true
```

Comment:
Could you please add a comment on why privileged is needed?

Reply:
privileged is needed to access /dev/accel/accel0-7 device nodes.

Reply:
> privileged is needed to access /dev/accel/accel0-7 device nodes.

it's a little weird as other cmds should also need to access those dev nodes.

Reply:
yes. need to find out how habana docker runtime manages this access issue but  we also have this privileged set as true for our XPU docker run.

Reply:
I've never used Docker compose, just plain docker and Kubernetes.

Kubernetes manifest files request device resources of specific types.

Whereas with plain docker, you ask for specific device file(s).  Docker will then add them to container's devices cgroup.

Container user or group ID needs to match host device file access rights, easiest is just running container with root ID.  But then one should at least drop all capabilities as those are not needed for device access.

Reply:
@eero-t 

all gaudi devices nodes are under root, so you won't be able to access those files with other user accounts.
![image](https://github.com/user-attachments/assets/8ad1b715-139e-4a96-b21c-74a8d168b38e)

To add devices, we might have 1 card or 8 cards. not sure how to add needed device files easily.  
In the meantime, privileged mode works easily for all cases.

Reply:
> yes. need to find out how habana docker runtime manages this access issue but we also have this privileged set as true for our XPU docker run.

Could someone help connect @louie-tsai to the Habana runtime folks to sort out this privilege issue. Not eager to create security holes...

Reply:
> To add devices, we might have 1 card or 8 cards. not sure how to add needed device files easily.

@louie-tsai At least with plain Docker, you don't need to give devices individually, you can also give all devices of given type, by specifying corresponding devices dir:
* gpu: `--device /dev/dri:/dev/dri:rw`
* gaudi(?):  `--device /dev/accel:/dev/accel:rw`

Reply:
that might work. Since there is concern to have privilege on, let me remove that first. maybe it is related to some issues for habana runtime on my gaudi3 env.